mining/src/mempool/mod.rs

```diff
@@ -0,0 +1,120 @@
+use crate::model::owner_txs::{OwnerSetTransactions, ScriptPublicKeySet};
+
+use self::{
+    config::Config,
+    model::{orphan_pool::OrphanPool, pool::Pool, transactions_pool::TransactionsPool},
+};
+use consensus_core::{
+    api::DynConsensus,
+    tx::{MutableTransaction, TransactionId},
+};
+use std::rc::Rc;
+
+pub(crate) mod check_transaction_standard;
+pub mod config;
+pub mod errors;
+pub(crate) mod handle_new_block_transactions;
+mod model;
+pub(crate) mod populate_entries_and_try_validate;
+pub(crate) mod remove_transaction;
+pub(crate) mod revalidate_high_priority_transactions;
+pub(crate) mod validate_and_insert_transaction;
+
+/// Mempool contains transactions intended to be inserted into a block and mined.
+///
+/// Some important properties to consider:
+///
+/// - Transactions can be chained, so a transaction can have parents and chained
+///   dependencies in the mempool.
+/// - A transaction can have some of its outpoints refer to missing outputs when
+///   added to the mempool. In this case it is considered orphan.
+/// - An orphan transaction is unorphaned when all its UTXO entries have been
+///   built or found.
+/// - There are transaction priorities: high and low.
+/// - Transactions submitted to the mempool by a RPC call have **high priority**.
+///   They are owned by the node, they never expire in the mempool and the node
+///   rebroadcasts them once in a while.
+/// - Transactions received through P2P have **low-priority**. They expire after
+///   60 seconds and are removed if not inserted in a block for mining.
+pub(crate) struct Mempool {
+    config: Rc<Config>,
```

Comment:
When `MiningManager` is used and passed between threads this line will not compile bcs of the Rc. It must be an Arc (same goes for Config in the pools within)